src/eliater/repair.py

```diff
@@ -0,0 +1,161 @@
+that there exists an additional bidirectional edge between M2 and Y, which is not depicted
+in the given ADMG. Our objective is to identify all the conditional independencies implied
+by this ADMG and pinpoint any inconsistencies with the data. The ultimate aim is to detect
+this overlooked bidirectional edge and incorporate it into the corrected ADMG.
+
+.. code-block:: python
+
+    from y0.graph import NxMixedGraph
+    from frontdoor_backdoor.multiple_mediators_single_confounder import generate
+    from eliater.repair import fix_graph
+
+    graph = NxMixedGraph.from_edges(
+        directed=[
+            ('X', 'M1'),
+            ('M1', 'M2'),
+            ('M2', 'Y'),
+        ],
+        undirected=[
+            ('X', 'Y'),
+        ],
+    )
+
+    # Generate observational data for this graph (this is a special example)
+    observational_data = generate(100)
+
+    repaired_graph = repair_network(graph, observational_data)
+
+"""
+
+import warnings
+from typing import Dict, Literal, Optional
+
+import pandas as pd
+
+from y0.algorithm.falsification import get_conditional_independencies
+from y0.dsl import Variable
+from y0.graph import NxMixedGraph
+from y0.struct import get_conditional_independence_tests
+
+__all__ = [
+    "repair_network",
+]
+
+
+def get_state_space_map(
+    data: pd.DataFrame, threshold: Optional[int] = 10
+) -> Dict[Variable, Literal["discrete", "continuous"]]:
+    """Get a dictionary from each variable to its type."""
+    column_values_unique_count = {
+        column_name: data[column_name].nunique() for column_name in data.columns
+    }
+    return {
+        Variable(column): "discrete"
+        if column_values_unique_count[column] <= threshold
+        else "continuous"
+        for column in data.columns
+    }
+
+
+def is_data_discrete(data: pd.DataFrame) -> bool:
+    """Check if all the columns in the dataframe has discrete data."""
+    variable_types = set(get_state_space_map(data=data).values())
+    return variable_types == {"discrete"}
+
+
+def is_data_continuous(data: pd.DataFrame) -> bool:
+    """Check if all the columns in the dataframe has continuous data."""
+    variable_types = set(get_state_space_map(data).values())
+    return variable_types == {"continuous"}
+
+
+CITest = Literal[
+    "pearson",
+    "chi-square",
+    "cressie_read",
+    "freeman_tuckey",
+    "g_sq",
+    "log_likelihood",
+    "modified_log_likelihood",
+    "power_divergence",
+    "neyman",
+]
+
+
+def choose_default_test(data: pd.DataFrame) -> CITest:
+    """Choose the default statistical test for testing conditional independencies based on the data."""
+    if is_data_discrete(data):
+        return "chi-square"
+    if is_data_continuous(data):
+        return "pearson"
+    raise NotImplementedError(
+        "Mixed data types are not allowed. Either all of the columns of data should be discrete / continuous."
+    )
+
+
+def repair_network(
```

Comment:
changing from `fix_graph` to `repair_network` did not address my concern of "this is not descriptive"

Reply:
Really brainstorm what this is doing from a high level. It's certainly not repairing, but in some way modifying the network. So take a big step back and thing about what are the high level goals of this function when coming up with a better name